tests/db/utils_test.py

```diff
@@ -189,3 +189,74 @@ def test_comma_in_names_are_allowed(self):
         actual = utils.encode_header('from', addresses)
         expected = email.header.Header(addresses)
         self.assertEqual(str(actual), str(expected))
+
+
+class TestDecodeHeader(unittest.TestCase):
+
+    def _test(self, teststring, expected):
+        actual = utils.decode_header(teststring)
+        self.assertEqual(actual, expected)
+
+    def test_non_ascii_strings_are_returned_as_unicode_directly(self):
+        text = u'Nön ÄSCII string¡'
+        self._test(text, text)
+
+    def test_basic_utf_8_quoted(self):
+        text = '=?utf-8?Q?=C3=84=C3=96=C3=9C=C3=A4=C3=B6=C3=BC?='
```

Comment:
These are difficult to read, is it possible to do something like: `'=?iso-8859-1?Q?' + expected.encode('iso-8859-1')`, or will that produce the wrong values?